### PR TITLE
DSS Collection repository review, refactor and cleanup

### PR DESCRIPTION
**Pull request by mimran-DFE**
DSS Collection repository review, refactor and cleanup

### CODE REVIEW

File: NCS.DSS.Collections/Cosmos/Provider/ComosDBProvider.cs
Comment (ajones-DFE) [inline NCS.DSS.Collections/Cosmos/Provider/ComosDBProvider.cs:1]: This file name spelling mistake needs correcting, No logging present currently

File: NCS.DSS.Collections/ServiceBus/ContentEnhancer/Client/ContentEnhancerServiceBusClient.cs
Comment (ajones-DFE) [inline NCS.DSS.Collections/ServiceBus/ContentEnhancer/Client/ContentEnhancerServiceBusClient.cs:1]: Logging required

File: NCS.DSS.Collections/ServiceBus/DataCollections/Client/DataCollectionsServiceBusClient.cs
Comment (ajones-DFE) [inline NCS.DSS.Collections/ServiceBus/DataCollections/Client/DataCollectionsServiceBusClient.cs:1]: Logging required

File: NCS.DSS.Collections/ServiceBus/DataCollections/Processor/Service/DataCollectionsQueueProcessorService.cs
Comment (ajones-DFE) [inline NCS.DSS.Collections/ServiceBus/DataCollections/Processor/Service/DataCollectionsQueueProcessorService.cs:1]: Need to also log steps not just exceptions. e.g. "Attempting to retrieve collection for id {id}" on line 43

File: NCS.DSS.Collections/ServiceBus/DataCollections/Processor/Service/DataCollectionsQueueProcessorService.cs [hunk 1]
```diff
@@ -1,52 +1,55 @@
...
+
+        public DataCollectionsQueueProcessorService(IDataCollectionsMessageProvider messageProvider,
+                                                    IContentEnhancerServiceBusClient contentEnhancerServiceBusClient,
+                                                    ICosmosDBProvider cosmosDbProvider,
+                                                    ILogger<DataCollectionsQueueProcessorService> logger)
+        {
+            _messageProvider = messageProvider;
+            _cosmosDbProvider = cosmosDbProvider;
+            _contentEnhancerServiceBusClient = contentEnhancerServiceBusClient;
            _logger = logger;
```
Comment (ajones-DFE) [inline NCS.DSS.Collections/ServiceBus/DataCollections/Processor/Service/DataCollectionsQueueProcessorService.cs:24]: Please remove this code highlighted in red

**Review by ajones-DFE — COMMENTED** [review]
Needs more in depth logging

File: NCS.DSS.Collections/Cosmos/Provider/ComosDBProvider.cs
Comment (mimran-DFE) [reply NCS.DSS.Collections/Cosmos/Provider/ComosDBProvider.cs:1]: This is fixed

**Review by ajones-DFE — APPROVED** [review]

**Pull request merged**